dipy/io/gradients.py

```diff
@@ -45,12 +44,21 @@ def read_bvals_bvecs(fbvals, fbvecs):
                    '.eddy_rotated_bvecs', '']:
             with open(this_fname, 'r') as f:
                 content = f.read()
-            # We replace coma and tab delimiter by space
-            with InTemporaryDirectory():
-                tmp_fname = "tmp_bvals_bvecs.txt"
-                with open(tmp_fname, 'w') as f:
-                    f.write(re.sub(r'(\t|,)', ' ', content))
-                vals.append(np.squeeze(np.loadtxt(tmp_fname)))
+
+            # Note: no context manager used because of windows behavior
+            # where tempfile not visible while the while is still open,
+            # so explicit cleanup required
+            f = tempfile.SpooledTemporaryFile(suffix=".txt", mode="w+")
+            try:
+                # We replace coma and tab delimiter by space
+                replaced_content = re.sub(r'(\t|,)', ' ', content)
+                f.write(replaced_content)
+                f.flush()
+                f.seek(0)
+                numpy_data = np.loadtxt(f)
+                vals.append(np.squeeze(numpy_data))
+            finally:
+                f.close()
```

Comment:
Sorry for hopping in late, but if you can `np.loadtxt()` on a filehandle, why not just use `StringIO`?

```suggestion
            munged_content = io.StringIO(re.sub(r'(\t|,)', ' ', content))
            vals.append(np.squeeze(np.loadtxt(munged_content)))
```